keps/sig-instrumentation/4827-component-statusz/README.md

```diff
@@ -0,0 +1,406 @@
+
+
+### Goals
+
+* Enhance observability into component health by providing a concise overview of critical component information, such as build version, Go version, compatibility details, and other pertinent data points
+* Set a precedent for presenting this basic but crucial component information in a consistent and accessible manner across the Kubernetes ecosystem
+
+### Non-Goals
+
+* Replace existing monitoring solutions: metrics, logs, traces
+* To provide a general-purpose extension mechanism for other components (e.g., container runtime, CNI, device plugins) to expose their own information through z-pages
+* Provide information about components that are not directly accessible due to network restrictions
+
+## Proposal
+
+We are proposing to add a new endpoint, /statusz on all core Kubernetes components which returns key details regarding a component's current state, including versioning, build information, and compatibility.
+
+
+### Risks and Mitigations
+
+1. **No sensitive data exposed**
+    
+    We will ensure that no sensitive data is exposed through statusz and that access to this endpoint is gated by using `system:monitoring group` ([ref](https://github.com/kubernetes/kubernetes/blob/release-1.31/staging/src/k8s.io/apiserver/pkg/authentication/user/user.go#L73)).
+
+2. **Performance impact**
+    
+    We will take care to carefully consider the computational overhead associated with the statusz endpoint to avoid negatively impacting the server's overall performance.
+
+3. **Essential data only**
+    
+    We will prioritize inclusion of only essential diagnostic and monitoring data that aligns with the intended purpose of the z-page.
+
+4. **Premature Dependency on Unstable Format**
+
+    The alpha release will explicitly support plain text format only, making it clear that the output is not intended for parsing or automated monitoring. The feature will be secured behind a feature gate that is disabled by default, ensuring users opt-in consciously. Also, the endpoint will be implemented with versioned URLs to facilitate future transitions to structured formats without breaking existing integrations, once the usage patterns and requirements are better understood
+
+## Design Details
+
+### Data Format and versioning
+
+Initially, the statusz page will exclusively support a plain text format for responses. We will implement these endpoints using versioned URLs, like /statusz/v1, to ensure future compatibility. This versioning strategy allows us to seamlessly introduce structured data formats (e.g., JSON) in the future, through a distinct endpoint such as /statusz/v2, without disrupting existing implementations.
+
+### Authz and authn
+
+Access to the endpoint will be limited to members of the `system:monitoring group` ([ref](https://github.com/kubernetes/kubernetes/blob/release-1.31/staging/src/k8s.io/apiserver/pkg/authentication/user/user.go#L73)), which is how paths like /healthz, /livez, /readyz are restricted today. 
```

Comment:
@vinayakankugoyal any feedback based on https://github.com/kubernetes/enhancements/issues/2862?

Reply:
Left a few comments! Thanks for bringing this KEP to my attention.